### PR TITLE
Add Wiki link to diaries

### PR DESCRIPTION
**Pull request by jzimmm**
Added Wiki links to Diary menus for quick access to achievement diary guides.

Things to note:
Achievement diaries do not have quick guides so I only added a regular guide option.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java [hunk 1]
```diff
@@ -262,27 +264,58 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		int widgetIndex = event.getActionParam0();
 		int widgetID = event.getActionParam1();
 
-		if (!Ints.contains(QUESTLIST_WIDGET_IDS, widgetID) || !"Read Journal:".equals(event.getOption()))
+		//check to see if mouse is pointing to a quest or a diary
+		boolean isQuest = Ints.contains(QUESTLIST_WIDGET_IDS, widgetID) || "Read Journal:".equals(event.getOption());
+		boolean isDiary = Ints.compare(DIARYLIST_WIDGET_ID, widgetID) == 0 && ("Open Ardougne Journal".equals(event.getOption())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java]: You can probably just regex match this (Open [\w &]+ Journal)

**Comment by deathbeam** [comment]
Use `Wiki ` instead of Guide for diaries (Guide and Quick Guide makes sense only for Quests, otherwise we should just use Wiki everywhere)

**Comment by jzimmm** [comment]
Just to clarify you just want  "Wiki" and nothing else?

**Comment by deathbeam** [comment]
well wiki and target.

**Comment by jzimmm** [comment]
Okay so like "Ardougne Diary Wiki"

**Comment by deathbeam** [comment]
no, there is menu option and menu target, and I think target is already ardougne dairy? so just wiki as menuop should be enough

**Comment by jzimmm** [comment]
Okay will do tonight, thanks.

**Comment by jzimmm** [comment]
Made the requested changes.

Here is what the dairy menu option looks like:
image

**Comment by jzimmm** [comment]
Fixed merge conflicts.

File: runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java [hunk 2]
```diff
@@ -246,18 +253,22 @@ private void onMenuOptionClicked(MenuOptionClicked ev)
...
+					if (isSkill)
+					{
+						Matcher skillRegex = WikiPlugin.SKILL_REGEX.matcher(Text.removeTags(ev.getMenuTarget()));
+
+						if (skillRegex.find())
+						{
+							LinkBrowser.browse(WIKI_BASE + "/w/" + URLEncoder.encode(skillRegex.group(1)) + "?" + UTM_PARAMS);
+						}
+					}
+					//Diaries will fallthrough;
```
Comment (WillThomas) [inline runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java]: Looks like a skill lookup will fall through here too and open 2 wiki pages? Is isSkill required? If skillRegex doesn't match this shouldn't do anything.

File: runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java [hunk 3]
```diff
@@ -275,23 +286,46 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		int widgetID = event.getActionParam1();
 		MenuEntry[] menuEntries = client.getMenuEntries();
 
-		if (Ints.contains(QUESTLIST_WIDGET_IDS, widgetID) && "Read Journal:".equals(event.getOption()))
+		//check to see if mouse is pointing to a quest or a diary
+		boolean isQuest = Ints.contains(QUESTLIST_WIDGET_IDS, widgetID) || "Read Journal:".equals(event.getOption());
```
Comment (WillThomas) [inline runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java]:
```suggestion
		boolean isQuest = Ints.contains(QUESTLIST_WIDGET_IDS, widgetID) && "Read Journal:".equals(event.getOption());
```

File: runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java [hunk 4]
```diff
@@ -192,6 +196,9 @@ private void onDeselect()
 	@Subscribe
 	private void onMenuOptionClicked(MenuOptionClicked ev)
 	{
+		//check to see if its a skill menu
+		boolean isSkill = ev.getMenuOption().startsWith("View");
```
Comment (WillThomas) [inline runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java]: This is probably better placed in the if (ev.getMenuAction() == MenuAction.RUNELITE) block where it's used. It's getting set everytime this method is called here even for unrelated MenuOptions

File: runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java (hunk 2 above)
Comment (jzimmm) [reply runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java]: Yes I forgot to add the break statement inside the if. Fixing it now

**Comment by jzimmm** [comment]
This is done and ready to be merged.

**Comment by jzimmm** [comment]
Okay I don't understand what you want me to change?

**Pull request closed without merging**